lib/Dialect/TTIR/Transforms/Passes.cpp

```diff
@@ -565,9 +565,12 @@ class TTIRLayoutDPSOperandsRewriter
 
     assert(op->template hasTrait<TTIROp::Trait>());
     bool modified = false;
+    int idx = -1;
```

Comment:
What's the rationale for skipping the first 3 operands here?

Reply:
This is one of the reasons why this PR isn't ready yet.

So the TTNN conv2d op itself moved the input, weights, and biases from host onto device. If those tensors are already on device when the op is called it causes errors. My understanding is that we want to stop moving op outputs on and off host between every op, so something will have to change about the TTNN Conv2D op.